jetty-infinispan/infinispan-common/src/main/java/org/eclipse/jetty/session/infinispan/SessionDataMarshaller.java

```diff
@@ -49,6 +80,50 @@ public String getTypeName()
         return "org_eclipse_jetty_session_infinispan.InfinispanSessionData";
     }
 
+    @Override
+    public InfinispanSessionData readObject(ObjectInput input) throws IOException, ClassNotFoundException
+    {
+        if (serializationContext != null)
+        {
+            ByteArrayOutputStream os = new ByteArrayOutputStream();
+            int bufferSize = 4 * 1024;
+            byte[] buffer = new byte[bufferSize];
+            int len = -1;
+            while ((len = input.read(buffer, 0, bufferSize)) != -1)
+            {
+                os.write(buffer, 0, len);
+            }
+
+            // invokes readFrom(ProtoStreamReader)
+            InfinispanSessionData data = ProtobufUtil.fromByteArray(serializationContext, os.toByteArray(),
```

Comment:
Also here I wanted to avoid the double copying and buffering of data, which is why I was suggesting having a common method that worked with both ObejectInput and ProtoSteamReader.